catalog-service/src/main/java/com/example/catalogservice/web/controllers/ProductController.java

```diff
@@ -87,33 +91,35 @@ public Mono<ResponseEntity<Boolean>> productExistsByProductCodes(
     }
 
     @PostMapping
-    public Mono<ResponseEntity<Product>> createProduct(@RequestBody @Valid ProductDto productDto) {
+    public Mono<ResponseEntity<ProductResponse>> createProduct(
+            @RequestBody @Valid ProductDto productDto) {
```

Comment:
productDTO is used for sending data to kafka, it is kept in common.dtos package as a work around for jsonconverter at consumer side in kafka.  So please create a new POJO as ProductRequest which is exact duplicate of productDTO, remove validation from productDTO.